php-checks/src/main/java/org/sonar/php/checks/SelfKeywordUsageCheck.java

```diff
@@ -51,10 +54,18 @@
   private static final String MESSAGE = "Use \"static\" keyword instead of \"self\".";
 
   private Deque<Boolean> isFinalClassStack = new ArrayDeque<>();
+
   private Deque<Set<String>> finalMethodsStack = new ArrayDeque<>();
 
+  private List<String> privateProperties;
+
+  private List<String> privateMethods;
+
   @Override
   public void visitClassDeclaration(ClassDeclarationTree tree) {
+    privateProperties = new LinkedList<>();
```

Comment:
@ivandalbosco I think we should make these lists (btw why not sets?) be elements of stack (like `isFinalClassStack` and `finalMethodsStack`). Due to that we are able to cover nested classes (see unit test file)